rdkPlugins/OOMCrash/source/OOMCrashPlugin.cpp

```diff
@@ -0,0 +1,246 @@
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "OOMCrashPlugin.h"
+/**
+ * Need to do this at the start of every plugin to make sure the correct
+ * C methods are visible to allow PluginLauncher to find the plugin
+ */
+REGISTER_RDK_PLUGIN(OOMCrash);
+
+/**
+ * @brief Constructor - called when plugin is loaded by PluginLauncher
+ *
+ * Do not change the parameters for this constructor - must match C methods
+ * created by REGISTER_RDK_PLUGIN macro
+ *
+ * Note plugin name is not case sensitive
+ */
+OOMCrash::OOMCrash(std::shared_ptr<rt_dobby_schema> &containerConfig,
+                             const std::shared_ptr<DobbyRdkPluginUtils> &utils,
+                             const std::string &rootfsPath)
+    : mName("OOMCrash"),
+      mContainerConfig(containerConfig),
+      mRootfsPath(rootfsPath),
+      mUtils(utils)
+{
+    AI_LOG_FN_ENTRY();
+
+    AI_LOG_FN_EXIT();
+}
+
+/**
+ * @brief Set the bit flags for which hooks we're going to use
+ *
+ * This plugin uses the postInstallation and postHalt hooks, so set those flags
+ */
+unsigned OOMCrash::hookHints() const
+{
+    return (
+        IDobbyRdkPlugin::HintFlags::PostInstallationFlag |
+	IDobbyRdkPlugin::HintFlags::PostHaltFlag);
+}
+
+/**
+ *  * @brief Dobby Hook - run in host namespace *once* when container bundle is downloaded
+ *   */
+bool OOMCrash::postInstallation()
+{
+    if (!mContainerConfig)
+    {
+        AI_LOG_WARN("Container config is null");
+        return false;
+    }
+
+    const std::string path = mContainerConfig->rdk_plugins->oomcrash->data->path;
+    if (!mUtils->mkdirRecursive(mRootfsPath + path.c_str(), 0755) && errno != EEXIST)
+    {
+        AI_LOG_ERROR("failed to create directory '%s' (%d - %s)", (mRootfsPath + path).c_str(), errno, strerror(errno));
+        return false;
+    }
+
+    if (!mUtils->mkdirRecursive(path.c_str(), 0755) && errno != EEXIST)
+    {
+        AI_LOG_ERROR("failed to create directory '%s' (%d - %s)", path.c_str(), errno, strerror(errno));
+        return false;
+    }
+
+    if (!mUtils->addMount(path, path, "bind", {"bind", "ro", "nodev","nosuid", "noexec" }))
+    {
+        AI_LOG_WARN("failed to add mount %s", path.c_str());
+        return false;
+    }
+    AI_LOG_INFO("OOMCrash postInstallation hook is running for container with hostname %s", mUtils->getContainerId().c_str());
+    return true;
+}
+
+/**
+ * @brief Dobby Hook - Run in host namespace when container terminates
+ */
+bool OOMCrash::postHalt()
+{
+    if (!mContainerConfig)
+    {
+        AI_LOG_WARN("Container config is null");
+        return false;
+    }
+    
+    bool status = true;
```

Comment:
Why is `status` pre-set to true? This means that if `mUtils->exitStatus` will be zero then `status` will not be updated and the `createFileForOOM` file will be created, even though no OOM was there. Set it initially to false. And also you can name this variable better i.e. `oomDetected` or something similar.